File: OrienGO-Backend/src/main/java/com/example/oriengo/model/dto/AdminResponseDTO.java
package com.example.oriengo.model.dto;

import com.example.oriengo.model.enumeration.AdminLevel;
import com.example.oriengo.model.enumeration.Department;
import lombok.*;

import java.time.LocalDateTime;

@Getter
@Setter
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class AdminResponseDTO {
    private Long id;
    private String firstName;
    private String lastName;
    private int age;
    private String gender;  // We'll map GenderType enum to String
    private String phoneNumber;
    private String email;
    private boolean enabled;
    private AdminLevel adminLevel;
    private Department department;
    private Long createdById;  // just ID of creator to avoid loading full object
    private LocalDateTime createdAt;
    private LocalDateTime updatedAt;
    private LocalDateTime lastSeen;
}

File: OrienGO-Frontend/src/app/model/dto/AdminResponse.dto.ts
// admin-response.dto.ts

import { AdminLevel } from "../enum/AdminLevel.dto";
import { Department } from "../enum/Department.enum";

export interface AdminResponseDTO {
  id: number;
  firstName: string;
  lastName: string;
  age: number;
  gender: string; // mapped from enum to string
  phoneNumber?: string;
  email: string;
  enabled: boolean;
  adminLevel: AdminLevel;
  department: Department;
  createdById: number;
  createdAt?: string;
  updatedAt?: string;
  lastSeen?: string;
}
